src/components/redux/books/books.js

```diff
@@ -0,0 +1,30 @@
+// define the action type
+const ADD_BOOKS = 'bookstore/books/ADD_BOOKS';
+const REMOVE_BOOKS = 'bookstore/books/REMOVE_BOOKS';
+
+// initial state
+const initialState = { books: [] };
+
+//  the action type to define the action creator
+
+export const addBookAction = (newBook) => (
+  { type: ADD_BOOKS, newBook }
+);
+export const removeBookAction = (id) => (
+  { type: REMOVE_BOOKS, id }
+);
+
+// Reducer
+const booksReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOKS:
+      return {
+        ...state,
+        books: [...state.filter((book) => book.id !== action.id)],
+      };
```

Comment:
- You have done great so far :+1: But there are a few requirements you need to complete before moving to the next milestone. :+1: :100: 
- [ ] You did well adding the action creators as required, but in your reducer, the action type of `ADD_BOOKS` seems to be removing a book and not adding, I'll suggest you rename or re-write this functionality correctly. :+1: :+1: :smile: 
- [ ] Also, in your Reducer, you are meant to handle the two action types `REMOVE_BOOKS` and `ADD_BOOKS`, but currently, you only have case for one action type, I'll suggest you add the other. :small_airplane: :wave:

Reply:
Done, included remove case and edited add case.